tests/test_clicks.py

```diff
@@ -0,0 +1,115 @@
+from poprox_storage.repositories.clicks import DbClicksRepository
+from poprox_concepts import Account
+from datetime import datetime
+
+import json
+import pytest
+import os
+from sqlalchemy import create_engine, text
+from uuid import uuid4
+
+db_password = os.environ.get("POPROX_DB_PASSWORD", "")
+db_port = os.environ.get("POPROX_DB_PORT", "")
+
+DEFAULT_PG_URL = f"postgresql://postgres:{db_password}@127.0.0.1:{db_port}/poprox"
+
+
+@pytest.fixture(scope="session")
+def pg_url():
+    """
+    Provides base PostgreSQL URL for creating temporary databases.
+    """
+    return os.getenv("CI_POPROX_PG_URL", DEFAULT_PG_URL)
+
+
+def test_get_active_survey(pg_url: str):
+    engine = create_engine(pg_url)
+    with engine.connect() as conn:
+        conn.execute(text("delete from clicks;"))
+        conn.execute(text("delete from newsletters;"))
+        conn.execute(text("delete from articles;"))
+
+        dummy_click_1 = uuid4()
+        dummy_click_2 = uuid4()
+        dummy_user_1 = uuid4()
+        dummy_user_2 = uuid4()
+        dummy_newsletter = uuid4()
+
+        dummy_article_1 = uuid4()
+        dummy_article_2 = uuid4()
+
+        query_0 = text(
+                f'''insert into accounts(account_id, email, status, source) values (:account_id, 'user-1@gmail.com', '', '');'''
+            )
```

Comment:
`query_0` looks a lot like `DbAccountRepository.create_new_account()`. Could you use that here?

Reply:
> Ah I see, but looks like create_new_account() function does not take in account_id as a param

If you needed that, you could make it an optional parameter to that function with a default value of `None` to maintain backward compatibility for existing callers. None of the repository code is set in stone or even particularly good (which I can say with confidence because I wrote most of it), so you should feel free to make changes where you need to. However...

> also it returns an Account object 🤔 I can see it might be able to replace the two account objects created in line 103, but again does that mean we might need to manually fetch the created Account object and its account_id for other following queries?

As the code is currently written, you need to know the account ids up front, then use them to create the database state you need, then create corresponding account objects—which works fine but is kind of a lot of work. If you use `create_new_account()` instead somewhere up here in the setup code (before line 100 or so), then:
* You wouldn't need to manually generate the account ids with `uuid4()` because they'll be auto-assigned by `create_new_account()` (so you probably don't actually need to pass the ids in)
* You can assign the result to `accounts` (which would replace lines 103-104, as you point out)
* You can pull the auto-assigned account ids from the `Account` objects with `accounts[0].account_id`/`accounts[1].account_id` when you need to use them to record clicks below

It'll do essentially the same thing, just at a higher level of abstraction so you won't have to think as hard or write so much SQL/code.